Introduce Gulp and use Modernizr

sergeylukin opened the pull request:
- use `gulp` and `webpack-stream` instead of webpack directly
  to build the project
- use `gulp-modernizr` task to automatically generate
  `modernizr-custom.js` based on tested features and
  use it instead of manual detection
- use `gulp-jshint` to lint JS

sergeylukin committed:
Introduce Gulp and use Modernizr

- use `gulp` and `webpack-stream` instead of webpack directly
  to build the project

- use `gulp-modernizr` task to automatically generate
  `modernizr-custom.js` based on tested features and
  use it instead of manual detection

- use `gulp-jshint` to lint JS

charlespeters commented:
thanks for adding this. before i merge it, quick question: if we do the custom gulp modernizr, how would it work using it as a node_module in another project? doesn't the custom gulp modernizr need to be built each time?

sergeylukin committed:
Keep and track custom Modernizr file in ./src

yukon-sergey commented:
You're totally right. Let's put it in the `./src` directory and track it in there so that no build step would be required when using it via `npm`. Check out commit b21784e I just added where I implemented just that.

sergeylukin commented:
You're totally right. Let's put it in the ./src directory and track it in there so that no build step would be required when using it via npm. Check out commit b21784e I just added where I implemented just that.

charlespeters commented:
working with the dist here and it's giving me png instead of webp in chrome http://codepen.io/charlespeters/full/XbLvoY

sergeylukin committed:
Wait for Modernizr to perform WebP test and only then run Thunder if test failes
